### DISCUSSION TITLE
Overloaded '<' operator in subsets.Range class

### DISCUSSION THREAD

**Discussion by meshtag** [spcl/dace#921] [Q&A]
I am trying to add support for memlet range overlap warning(in context of #424) and my solution involves sorting a list of elements of type `subsets.Range`. For using python's inbuilt sorting function, I can overload '<' operator for  `subsets.Range` class or provide a custom compare function specifically for that call site. Which one is more feasible?

**Comment by tbennun** [comment]
Great idea, thanks. I see two issues with the check as it is implemented right now:
1. there needs to be some exception raised if the ranges are symbolic and cannot be compared (this is likely already happening)
2. Since comparing 2-tuples is not well defined, I personally think that ranges should be compared by their starting element and then their ending element, rather than end element only.

**Reply by meshtag** [reply]
For the second point, do you mean something like this : 
```
def __lt__(self, other):
        if (self.min_element() < other.min_element()):
            return 1
        elif (self.min_element() > other.min_element()):
            return 0
        else:
            if (self.max_element() < other.max_element()):
                return 1
            else:
                return 0
```

**Reply by tbennun** [reply]
yes, though [comparisons should return boolean values](https://docs.python.org/3/reference/expressions.html#comparisons).
